testing/scenario_app/ios/Scenarios/Scenarios/AppDelegate.m

```diff
@@ -19,31 +19,55 @@ - (BOOL)application:(UIApplication*)application
 
   // This argument is used by the XCUITest for Platform Views so that the app
   // under test will create platform views.
-  if ([[[NSProcessInfo processInfo] arguments] containsObject:@"--platform-view"]) {
-    FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"PlatformViewTest" project:nil];
-    [engine runWithEntrypoint:nil];
-
-    FlutterViewController* flutterViewController =
-        [[NoStatusBarFlutterViewController alloc] initWithEngine:engine nibName:nil bundle:nil];
-    [engine.binaryMessenger
-        setMessageHandlerOnChannel:@"scenario_status"
-              binaryMessageHandler:^(NSData* _Nullable message, FlutterBinaryReply _Nonnull reply) {
-                [engine.binaryMessenger
-                    sendOnChannel:@"set_scenario"
-                          message:[@"text_platform_view" dataUsingEncoding:NSUTF8StringEncoding]];
-              }];
-    TextPlatformViewFactory* textPlatformViewFactory =
-        [[TextPlatformViewFactory alloc] initWithMessenger:flutterViewController.binaryMessenger];
-    NSObject<FlutterPluginRegistrar>* registrar =
-        [flutterViewController.engine registrarForPlugin:@"scenarios/TextPlatformViewPlugin"];
-    [registrar registerViewFactory:textPlatformViewFactory withId:@"scenarios/textPlatformView"];
-    self.window.rootViewController = flutterViewController;
-  } else {
+  // The launchArgsMap should match the one in the `PlatformVieGoldenTestManager`.
+  static NSDictionary* launchArgsMap;
```

Comment:
`application:didFinishLaunchingWithOptions:` will only be called once, so you don't need the overhead of the static and dispatch_once.
Also, always use generics.
```
NSDictionary<NSString*, NSString*>* launchArgsMap;
```